File: app/models/evento.rb
class Evento < ApplicationRecord
  belongs_to :tipo_evento, required: true
  belongs_to :ativo, foreign_key: :ativo_id, required: true

  validates :data, presence: true
  validates :fator, numericality: true
  validates :valor, numericality: true, allow_nil: true

  def self.from_json(conteudo_json)
    evento = new

    evento.ativo = Ativo.find_by(codigo: conteudo_json['codigo_ativo'])
    evento.tipo_evento = TipoEvento.find_by(nome: conteudo_json['tipo'])
    evento.fator = conteudo_json['fator']
    evento.data = conteudo_json['data']
    evento.valor = conteudo_json['valor']

    evento
  end

  # rails_admin do
  #   configure :data do
  #     strftime_format '%d/%m/%Y'
  #   end
  # end
end

File: Stocks/Models/Evento.cs
using System.ComponentModel.DataAnnotations.Schema;
using Microsoft.EntityFrameworkCore;
using Stocks.Data;

namespace Stocks.Models;

/// <summary>
/// Representa um evento financeiro associado a um ativo.
/// Um evento pode ser um desdobramento, grupamento, bonificação, etc.
/// </summary>
public partial class Evento
{
    public int Id { get; set; }

    public DateTime Data { get; set; }

    public int Fator { get; set; }

    public decimal? Valor { get; set; }

    public int TipoEventoId { get; set; }

    public int AtivoId { get; set; }

    public virtual Ativo Ativo { get; set; }

    public virtual TipoEvento TipoEvento { get; set; }

    [NotMapped]
    public string CodigoAtivo { get; set; }

    [NotMapped]
    public string Tipo { get; set; }

    [NotMapped]
    public string DataEvento { get; set; }

    /// <summary>
    /// Completa os campos necessários da entidade Evento.
    /// </summary>
    /// <param name="db"></param>
    public async void CompletarCampos(BancoContext db)
    {
        Data = DateTime.Parse(DataEvento);

        TipoEvento = await db.TiposEvento.SingleAsync(t => t.Nome == Tipo);
        Ativo = await db.Ativos.SingleAsync(a => a.Codigo == CodigoAtivo);
    }
}
